internal/generate/clusterserviceversion/bases/clusterserviceversion.go

```diff
@@ -0,0 +1,199 @@
+	}
+	if b.Description == "" {
+		b.Description = b.DisplayName + " description. TODO."
+	}
+	if b.Maturity == "" {
+		b.Maturity = "alpha"
+	}
+	if b.Capabilities == "" {
+		b.Capabilities = "Basic Install"
+	}
+	if len(b.Keywords) == 0 || b.Keywords[0] == "" {
+		b.Keywords = []string{b.OperatorName}
+	}
+	if len(b.Links) == 0 || b.Links[0] == (v1alpha1.AppLink{}) {
+		b.Links = []v1alpha1.AppLink{
+			{
+				Name: b.DisplayName,
+				URL:  fmt.Sprintf("https://%s.domain", b.OperatorName),
+			},
+		}
+	}
+	if len(b.Icon) == 0 {
+		b.Icon = make([]v1alpha1.Icon, 1)
+	}
+	if b.Provider == (v1alpha1.AppLink{}) {
+		b.Provider = v1alpha1.AppLink{
+			Name: "Provider Name",
+			URL:  "https://your.domain",
+		}
+	}
+	if len(b.Maintainers) == 0 || b.Maintainers[0] == (v1alpha1.Maintainer{}) {
+		b.Maintainers = []v1alpha1.Maintainer{
+			{
+				Name:  "Maintainer Name",
+				Email: "your@email.com",
+			},
+		}
+	}
+}
+
+// makeNewBase returns a base v1alpha1.ClusterServiceVersion to modify.
+func (b ClusterServiceVersion) makeNewBase() *v1alpha1.ClusterServiceVersion {
+	return &v1alpha1.ClusterServiceVersion{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: v1alpha1.ClusterServiceVersionAPIVersion,
+			Kind:       v1alpha1.ClusterServiceVersionKind,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      b.OperatorName + ".vX.Y.Z",
+			Namespace: "placeholder",
+			Annotations: map[string]string{
+				"capabilities": b.Capabilities,
+				"alm-examples": "[]",
+			},
+		},
+		Spec: v1alpha1.ClusterServiceVersionSpec{
+			DisplayName: b.DisplayName,
+			Description: b.Description,
+			Provider:    b.Provider,
+			Maintainers: b.Maintainers,
+			Links:       b.Links,
+			Maturity:    b.Maturity,
+			Keywords:    b.Keywords,
+			Icon:        b.Icon,
+			InstallModes: []v1alpha1.InstallMode{
+				{Type: v1alpha1.InstallModeTypeOwnNamespace, Supported: true},
+				{Type: v1alpha1.InstallModeTypeSingleNamespace, Supported: true},
+				{Type: v1alpha1.InstallModeTypeMultiNamespace, Supported: false},
+				{Type: v1alpha1.InstallModeTypeAllNamespaces, Supported: true},
+			},
+		},
+	}
+}
+
+// readClusterServiceVersionBase returns the ClusterServiceVersion base at path.
+// If no base is found, readClusterServiceVersionBase returns an error.
+func readClusterServiceVersionBase(path string) (*v1alpha1.ClusterServiceVersion, error) {
+	b, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	scanner := k8sutil.NewYAMLScanner(bytes.NewBuffer(b))
+	for scanner.Scan() {
+		manifest := scanner.Bytes()
+		typeMeta, err := k8sutil.GetTypeMetaFromBytes(manifest)
+		if err != nil {
+			log.Debugf("Skipping non-Object manifest %s: %v", path, err)
+			continue
+		}
+		if typeMeta.Kind == v1alpha1.ClusterServiceVersionKind {
+			csv := &v1alpha1.ClusterServiceVersion{}
+			if err := yaml.Unmarshal(manifest, csv); err != nil {
+				return nil, fmt.Errorf("error unmarshalling ClusterServiceVersion from manifest %s: %w", path, err)
```

Comment:
Nit: curious as to why we use `%w` as opposed to `%v` here.

Reply:
I was thinking of being smart with decode errors for more informative error reporting but decided to hold off for this PR. They can be switched back to `%v`.